FeedKit/images/ImageRepository.swift

```diff
@@ -287,13 +297,14 @@ extension ImageRepository: Images {
     let req = ImageRepository.makeImageRequest(identifier: id)
     let blocker = DispatchSemaphore(value: 0)
```

Comment:
@kean While I have you here 😉 Can you think of a better way to load an image synchronously with Nuke?

Reply:
It’s for `MPNowPlayingInfoCenter` in case you’re wondering.

Reply:
Semaphores are fine, just make sure you only use this method [from the background thread](https://github.com/kean/FutureX/blob/0af4b26e0ef7424dff914605b50da754e36a8481/Sources/FutureExtensions.swift#L187). If you block the main thread, the completion closures are executed on the main thread, if it's blocked they won't get a chance to. It looks like it might be worth allowing users to configure the dispatch queue on which to execute completion closures.

I could add an [`isSynchronous`](https://developer.apple.com/documentation/photokit/phimagerequestoptions/1616945-issynchronous) to `ImageRequestOptions`.

Reply:
Cool, thanks. MediaPlayer calls this method within a block it submits to a dedicated queue `com.apple.MediaPlayer.MPNowPlayingInfoCenter/accessQueue`.

I think this use case is too niche for an option.